packages/gestalt/src/stackDisallowedProps.js

```diff
@@ -0,0 +1,80 @@
+// @flow strict
+
+const disallowedProps = [
+  // Default
+  'onClick',
+  'className',
+  'style',
+  // Extra Box props
+  'dangerouslySetInlineStyle',
```

Comment:
this…isn't great. but the alternative is switching to an allowlist, which would involve auditing every non-standard prop currently passed to Box. though verbose, this seems like the better solution